### PR TITLE
[interfaces]: Add support for multiple IP interfaces on eth0

### PR DESCRIPTION
**Pull request by stcheng**
- In minigraph.py, add general logic for parsing both loopback IP interfaces
  and management IP interfaces. Add support for both IPv4 and IPv6 interfaces
  on management interface.

- Update interfaces.j2 template and the corresponding test and sample output.

### CODE REVIEW

File: files/image_config/interfaces/interfaces.j2 [hunk 1]
```diff
@@ -20,11 +20,13 @@ iface lo {{ 'inet' if minigraph_lo_interface['addr'] | ipv4 else 'inet6' }} stat
 {% block mgmt_interface %}
 # The management network interface
 auto eth0
-{% if minigraph_mgmt_interface['addr'] %}
+{% if minigraph_mgmt_interfaces | length %}
```
Comment (lguohan) [inline files/image_config/interfaces/interfaces.j2]: there are quite a few other places using minigraph_mgmt_interface, need to change as well

'''
lgh@lgh-issaquah:/data/sonic/sonic-buildimage$ git grep minigraph_mgmt_interface
dockers/docker-fpm-frr/zebra.conf.j2:ip route 0.0.0.0/0 {{ minigraph_mgmt_interface['gwaddr'] }} 200
dockers/docker-fpm-gobgp/zebra.conf.j2:ip route 0.0.0.0/0 {{ minigraph_mgmt_interface['gwaddr'] }} 200
dockers/docker-fpm-quagga/zebra.conf.j2:ip route 0.0.0.0/0 {{ minigraph_mgmt_interface['gwaddr'] }} 200
dockers/docker-snmp-sv2/snmpd.conf.j2:{% if minigraph_mgmt_interface.addr | ipv4 %}
dockers/docker-snmp-sv2/snmpd.conf.j2:agentAddress udp:{{ minigraph_mgmt_interface.addr }}:161
files/image_config/interfaces/interfaces.j2:{% if minigraph_mgmt_interface['addr'] %}
files/image_config/interfaces/interfaces.j2:    address {{ minigraph_mgmt_interface['addr'] }}
files/image_config/interfaces/interfaces.j2:    netmask {{ minigraph_mgmt_interface['mask'] }}
files/image_config/interfaces/interfaces.j2:    up ip route add default via {{ minigraph_mgmt_interface['gwaddr'] }} dev eth0 table default
files/image_config/interfaces/interfaces.j2:    up ip rule add from {{ minigraph_mgmt_interface['addr'] }}/32 table default
files/image_config/interfaces/interfaces.j2:    down ip route delete default via {{ minigraph_mgmt_interface['gwaddr'] }} dev eth0 table default
files/image_config/interfaces/interfaces.j2:    down ip rule delete from {{ minigraph_mgmt_interface['addr'] }}/32 table default
files/image_config/ntp/ntp.conf.j2:interface listen {{ minigraph_mgmt_interface.addr }}
src/sonic-config-engine/minigraph.py:    results['minigraph_mgmt_interface'] = mgmt_intf
src/sonic-config-engine/minigraph.py:    results['minigraph_mgmt_interface'] = mgmt_intf
src/sonic-config-engine/tests/test_cfggen.py:        argument = '-v "minigraph_mgmt_interface[\'addr\']" -M "' + self.sample_device_desc + '"'
'''

Comment (stcheng) [inline files/image_config/interfaces/interfaces.j2]: is working on that
btw, does IPv6 have a gateway IP?

**Comment by stcheng** [comment]
another question, are we using only ipv4 for ntp/snmp?

**Comment by stcheng** [comment]
i'll close this pull request. maybe it's better to keep the previous attribute name and just add a new ipv6 one.

**Comment by lguohan** [comment]
let just ignore v6 mgmt. ip in case minigraph has it. Just need to deal with v4 good now.

**Pull request closed without merging**